### PR TITLE
Setup Xcode provisioning through variables and switch to channels

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

- Setup a pipeline variable we can use to provision the specific version of XCode we want to use
- Switch provisionator file to work off of stable/release channels when provisioning sdk stuff

### Testing Procedure ###
verify builds 

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: build/provisioning/provisioning.csx [hunk 1]
```diff
@@ -7,73 +7,73 @@ string androidSDK_windows = "";//"https://download.visualstudio.microsoft.com/do
...
 
 
 if (IsMac)
 {
-	// Item (XreItem.Xcode_11_4_0).XcodeSelect ();
+    
+	ForceJavaCleanup();
+	Item (XreItem.Java_OpenJDK_1_8_0_25);
+
+	string releaseChannel = Environment.GetEnvironmentVariable ("releaseChannel");
```
Comment (rmarinho) [inline build/provisioning/provisioning.csx:29]: Shouldn't this be `CHANNEL`

File: build/steps/build-osx.yml [hunk 2]
```diff
@@ -2,12 +2,24 @@ steps:
   - checkout: self
     clean: true
 
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision Xcode'
+    condition: and(ne(variables['REQUIRED_XCODE'], ''), eq(variables['buildForVS2017'], 'false'))
```
Comment (rmarinho) [inline build/steps/build-osx.yml:7]: What would this variable specify?

File: build/provisioning/provisioning.csx (hunk 1 above)
Comment (PureWeen) [reply build/provisioning/provisioning.csx:29]: I was thinking <What>Channel is a useful name. Maybe we get some other type of Channel? maybe we can call this one VisualStudioChannel?

File: build/steps/build-osx.yml (hunk 2 above)
Comment (PureWeen) [reply build/steps/build-osx.yml:7]: The enumeration from provisionator so like `Xcode_11_4_1`

https://github.com/xamarin/Xamarin.Forms/pull/10392/files#diff-5c1673f0212d0e8bde062ec627acc073R5

File: build/provisioning/provisioning.csx (hunk 1 above)
Comment (PureWeen) [reply build/provisioning/provisioning.csx]: alright @rmarinho changed this to just use CHANNEL variable

**Pull request merged**